### PR TITLE
Drone 0.5 support

### PR DESCRIPTION
**Pull request by frapposelli**
This PR adds Drone 0.5 support, some changes have been applied:

- The Drone token is now passed via the `DOWNSTREAM_TOKEN` secret.
- The `server` directive must be specified every time as I couldn't reproduce the same behavior on the new API.
- Dependencies are now vendored using `govend`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,14 +7,11 @@ COMMIT ?= $(shell git rev-parse --short HEAD)
 LDFLAGS = -X "main.buildCommit=$(COMMIT)"
```
Comment (tboerger) [inline Makefile]: The makefile should be dropped

File: main.go [hunk 2]
```diff
@@ -3,109 +3,59 @@ package main
...
+		},
+		cli.StringFlag{
+			Name:   "server",
+			Usage:  "trigger a drone build on a custom server",
+			EnvVar: "PLUGIN_SERVER",
+		},
+		cli.StringFlag{
+			Name:   "token",
+			Usage:  "drone API token from your user setttings",
+			EnvVar: "DOWNSTREAM_TOKEN",
```
Comment (tboerger) [inline main.go]: Please add also PLUGIN_TOKEN, otherwise the token can be set only via secrets

File: vendor/github.com/drone/drone-go/LICENSE [hunk 3]
```diff
@@ -0,0 +1,202 @@
+Apache License
```
Comment (tboerger) [inline vendor/github.com/drone/drone-go/LICENSE]: Take a proper copy, this line should be centered

**Comment by tboerger** [comment]
Have you executed govend -vtul --prune?

**Comment by tboerger** [comment]
Beside the above comments I have created prs like https://github.com/drone-plugins/drone-slack/pull/18 to update the 0.5 branches

File: vendor/github.com/drone/drone-go/LICENSE (hunk 3 above)
Comment (frapposelli) [inline vendor/github.com/drone/drone-go/LICENSE]: This comes from a vendored package, can't change it.

**Comment by frapposelli** [comment]
@tboerger should have addressed all the points (except for the license), let me know if you prefer to squash all the commits.

**Comment by tboerger** [comment]
Yes, drop the last commit, I have used govendor for vendoring as brad used it multiple times because it got cleaner results :)

Beside that it looks totally fine.

File: vendor/github.com/drone/drone-go/LICENSE (hunk 3 above)
Comment (tboerger) [inline vendor/github.com/drone/drone-go/LICENSE]: You are right of course

**Comment by frapposelli** [comment]
@tboerger dropped the last commit, should I squash as well?

**Comment by frapposelli** [comment]
@tboerger any updates?

**Comment by tboerger** [comment]
@donny-dont and me will merge various prs within the next days.

**Comment by donny-dont** [comment]
LGTM

**Comment by nlf** [comment]
LGTM

**Pull request merged**